### ISSUE TITLE
Don't populate keepass.ini with KeeShare settings when KeeShare isn't used

### ISSUE THREAD

**Issue by librick** [keepassxreboot/keepassxc#9601]
## Summary
I use KeePassXC; I don't use the KeeShare functionality.
I also use the ~/.config/keepassxc/keepassxc.ini config file to manage my KeePassXC application settings and provide portability between machines.

By default, this file has a `[KeeShare]` section with properties `Active` and `Own`, each of which containing verbose XML (with `Own` in particular containing a private key). I don't use KeeShare, yet these two properties significantly add to the length of the keepassxc.ini file.

## Context and Proposed Change
See my related issue on the UX of the KeeShare settings page: https://github.com/keepassxreboot/keepassxc/issues/9600. Imo, in addition to the existing KeeShare-related settings, the keepassxc.ini file should contain a line explicitly enabling/disabling KeeShare (e.g., `Enabled=true` or something similar). A single checkbox in the KeeShare settings would control this value.

If KeeShare has not been enabled, do not populate `keepassxc.ini` with KeeShare-specific properties. These properties should only be enabled *after* KeeShare has been explicitly enabled with a single checkbox. This would make `keepassxc.ini` files more readable for users who don't use KeeShare.

## Version
KeePassXC 2.7.4
Debian 12

**Comment by librick** [comment]
Just tried a quick experiment. I closed KeePassXC, then deleted the `Own` property from my `~/.config/keepassxc/keepassxc.ini` file. Then I opened KeePassXC and monitored the contents of that file. As soon as I opened the application settings menu (just by clicking the settings icon/cog button), the `Own` property was generated. So even when I don't use KeeShare and don't have KeeShare-related settings in my `keepassxc.ini` file, my config file gets populated with KeeShare-specific info as soon as I open the applications settings menu. I didn't even click the KeeShare settings tab specifically.
